extensions/ql-vscode/src/view/result-tables.tsx

```diff
@@ -76,14 +75,26 @@ export class ResultTables
   }
 
   render(): React.ReactNode {
-    const selectedTable = this.state.selectedTable;
+    const { selectedTable } = this.state;
     const resultSets = this.getResultSets();
     const { database, resultsPath, kind } = this.props;
 
     // Only show the Problems view display checkbox for the alerts table.
-    const toggleDiagnosticsClass = cx(toggleDiagnosticsClassName, {
-      [toggleDiagnosticsSelectedClassName]: selectedTable === ALERTS_TABLE_NAME
-    });
+    const diagnosticsCheckBox = selectedTable === ALERTS_TABLE_NAME ?
```

Comment:
I don't care for the fact that
```js
  let val: SomeType | undefined = undefined;
  if (condition)
    val = someComputation();
```
requires a big messy type annotation. Would you be ok with the idiom `condition && someComputation()` instead of the ternary, since at least in that case there isn't an `undefined` stranted after the `someComputation()`? (I didn't realize until I looked it up just now that react also renders `false` as nothing, just as it does `undefined`)

(A utility function `renderIf<T>(condition: boolean, value: T): T | undefined` could work, though it would have the undesirable consequence that `value` would be computed regardless of the condition, and 
```
renderIf<T>(condition: boolean, computation: () => T): T | undefined {
  if (condition) return value(); else return undefined;
}
```
but that would make you have to throw extra `() =>` in every callsite)

Reply:
(@henrymercer agreed in an earlier conversation that it's a tragedy typescript lacks good concrete syntax for conditionals and switch statements as expressions :)

Reply:
Sigh. The minor increase in readability is not worth any of these :) I do appreciate the thorough investigation of options.